Node/src/TeamsChatConnector.ts

```diff
@@ -167,21 +175,29 @@ export class TeamsChatConnector extends builder.ChatConnector {
       // Filter out @mention
       var out: builder.IEvent[] = []; 
       for (var re of realEvents) {
-        let intentMessage = <IntentMessage>re;
+        let intentMessage = <builder.IMessage>re;
```

Comment:
This look strange here, move into the if block after we check that it's of type message.